confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/macros/MacroToContentConverter.java

```diff
@@ -0,0 +1,79 @@
+package org.xwiki.contrib.confluence.filter.internal.macros;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.contrib.confluence.filter.input.ConfluenceInputContext;
+import org.xwiki.contrib.confluence.filter.input.ConfluenceInputProperties;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.listener.Listener;
+import org.xwiki.rendering.parser.ParseException;
+import org.xwiki.rendering.parser.Parser;
+import org.xwiki.rendering.syntax.Syntax;
+
+/**
+ * Converts a macro call to a group, retaining the id, class and content.
```

Comment:
I'm wondering if it would not worth moving that work to the parser for once, through a list of macros in a ConfluenceXHTMLInputProperties's property, to reduce the amount of parsing/rendering required.

Reply:
> Yep, it's probably also a bit more straightforward?

It would be a much simpler code, yes.

Reply:
I also think it's worth making the change. I didn't really like the idea of parsing the content a second time

Reply:
I created a ticket in this regard: https://jira.xwiki.org/browse/CONFLUENCE-274 

I also started working on it here https://github.com/trrenty/confluence/tree/CONFLUENCE-274 . There are still some things I have to figure out, but I can't spend any more time on it right now.

Ill commit these converters for now